src/ArrayCollection.php

```diff
@@ -65,6 +71,7 @@ class ArrayCollection implements Collection, Selectable, Stringable
     public function __construct(array $elements = [])
     {
         $this->elements = $elements;
+        $this->size = count($elements);
```

Comment:
This increases the cost of the constructor call from O(1) to O(n).

Reply:
only if you create a collection from another collection.
All new collection costs O(1)

Reply:
> only if you create a collection from another collection.

Which is a totally valid thing to do! This is a blocker!